### PR TITLE
feat(sanity-check): check portal config works for homepage charts

### PR DESCRIPTION
**Pull request by qingyashu**
Before starting the server, let's check if the homepage chart config in portal config is correct. 

### Improvements
- Before starting, check if the homepage chart config in portal config is correct

### CODE REVIEW

File: sanity-check.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+/**
+ * this transfer '_xxx_count' to 'xxx'
+ * @param {string} countName
+ */
+function parseNodeType(countName) {
+  return countName.substring(1, countName.length - '_count'.length);
+}
+const boardCountNodes = params.config.graphql.boardCounts.map(e => parseNodeType(e.graphql));
+const chartCountNodes = params.config.graphql.boardCounts.map(e => parseNodeType(e.graphql));
```
Comment (mfshao) [inline sanity-check.js]: should be `graphql.chartCounts` ?

Comment (qingyashu) [reply sanity-check.js]: 🤔  why? it's getting the node name from chartCounts config....

Comment (qingyashu) [reply sanity-check.js]: Oh I get it....typo here !!

File: sanity-check.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ */
+function checkNodesExistInDict(nodeList) {
+  for (let i = 0; i < nodeList.length; i += 1) {
+    const nodeID = nodeList[i];
+    if (typeof dict[nodeID] === 'undefined') {
+      console.error(`ERR: "${nodeID}" does not exist`);
+      process.exit(1);
+    }
+  }
+  return true;
```
Comment (mfshao) [inline sanity-check.js]: is this return necessary?

**Pull request merged**